cli/module_loader.rs

```diff
@@ -134,30 +137,52 @@ impl CliModuleLoader {
     maybe_referrer: Option<ModuleSpecifier>,
   ) -> Result<ModuleSource, AnyError> {
     let code_source = if self.ps.npm_resolver.in_npm_package(specifier) {
-      let file_path = specifier.to_file_path().unwrap();
+      let is_cjs = self.ps.cjs_resolutions.lock().contains(specifier);
+      let (maybe_translate_kind, load_specifier) = if is_cjs {
+        if let Some(new_path) =
+          specifier.path().strip_suffix(node::CJS_TO_ESM_NODE_SUFFIX)
+        {
+          let mut specifier = specifier.clone();
+          specifier.set_path(new_path);
+          (Some(CjsToEsmTranslateKind::Node), Cow::Owned(specifier))
+        } else if let Some(new_path) =
+          specifier.path().strip_suffix(node::CJS_TO_ESM_DENO_SUFFIX)
+        {
+          let mut specifier = specifier.clone();
+          specifier.set_path(new_path);
+          (Some(CjsToEsmTranslateKind::Deno), Cow::Owned(specifier))
```

Comment:
```suggestion
        let mut specifier = specifier.clone();
        let path = specifier.path();
        if let Some(new_path) =
          path.strip_suffix(node::CJS_TO_ESM_NODE_SUFFIX)
        {
          specifier.set_path(new_path);
          (Some(CjsToEsmTranslateKind::Node), Cow::Owned(specifier))
        } else if let Some(new_path) =
         path.strip_suffix(node::CJS_TO_ESM_DENO_SUFFIX)
        {
          specifier.set_path(new_path);
          (Some(CjsToEsmTranslateKind::Deno), Cow::Owned(specifier))
```

Reply:
if that compiles :). LGTM otherwise

Reply:
I wish the url crate had "with" methods, so you could do `url.with_path(path)` to create a new url with that path.